src/Compilers/CSharp/Portable/Symbols/NamespaceSymbol.cs

```diff
@@ -377,6 +395,17 @@ ImmutableArray<INamespaceSymbol> INamespaceSymbol.ConstituentNamespaces
             }
         }
 
+        // PERF: initialization will allocate, so we make this lazy
+        private string _lazyQualifiedName;
```

Comment:
Please move field declarations to the beginning of the type declaration, before other members.